models/ptf.py

```diff
@@ -38,10 +58,111 @@ def get_filename_regex(cls):
+        The section_id must be a valid section identifier ([0, 11]).
+
+        Returns
+        -------
+        section: SensorSection
+            A new section for this instrument.
+        """
+        (dx, dy) = self.get_section_offsets(section_id)
+        defective = section_id == 1  # TODO: check which chip is defective!
+        return SensorSection(section_id, self.name, size_x=2048, size_y=4096,
+                             offset_x=dx, offset_y=dy, defective=defective)
+
+    def _get_default_calibrator(self, mjd, section, calibtype='dark', filter=None, session=None):
+        pass
+
+    @classmethod
+    def GaiaDR3_to_instrument_mag(cls, filter, catdata):
+        """Transform Gaia DR3 magnitudes to instrument magnitudes.
+
+        Uses a polynomial transformation from Gaia MAG_G to instrument magnitude.
+
+        The array trns allows a conversion from Gaia MAG_G to
+        the magnitude through the desired filter using:
+
+          MAG_filter = Gaia_MAG_G - sum( trns[i] * ( Gaia_MAG _BP - Gaia_MAG_RP ) ** i )
+
+        (with i running from 0 to len(trns)-1).
+
+        Parameters
+        ----------
+        filter: str
+            The (short) filter name of the magnitudes we want.
+        catdata: dict or pandas.DataFrame or numpy.recarray or astropy.Table
+            A data structure that holds the relevant data,
+            that can be indexed on the following keys:
+            MAG_G, MAGERR_G, MAG_BP, MAGERR_BP, MAG_RP, MAGERR_RP
+            If a single magnitude is required, can pass a dict.
+            If an array of magnitudes is required, can be any
+            data structure that when indexed on those keys
+            returns a 1D numpy array (e.g., a pandas DataFrame,
+            or a named structured numpy array, or even a dict
+            with ndarray values).
+
+        Returns
+        -------
+        trans_mag: float or numpy array
+            The catalog magnitude(s) transformed to instrument magnitude(s).
+        trans_magerr: float or numpy array
+            The catalog magnitude error(s) transformed to instrument magnitude error(s).
         """
-        return 'PTF'
+        if not isinstance(filter, str):
+            raise ValueError(f"The filter must be a string. Got {type(filter)}. ")
+
+        # Emily Ramey came up with these by fitting polynomials to Gaia
+        # magnitudes and DECaLS magnitudes
+        # transformations = {
+        #     'g': np.array([0.07926061, -0.18958323, -0.50588824, 0.11476034]),
+        #     'r': np.array([-0.28526417, 0.65444024, -0.25415955, -0.00204337]),
+        #     'i': np.array([-0.2491122, 0.51709843, 0.02919352, -0.02097517]),
+        #     'z': np.array([-0.38939061, 0.70406435, 0.04190059, -0.01617815])
+        # }
+        # TODO: must find better transformations!
+        transformations = {
+            'R': np.array([0]),
+            'V': np.array([0]),
+            'I': np.array([0]),
+        }
```

Comment:
...I might say "must find transformations" rather than "must find better transformations" in the TODO, because it's very obviously just a placeholder right now, and will not have any resemblance to reality.

(Somebody somewhere must have done this already.)